spec/main_spec.rb

```diff
@@ -0,0 +1,217 @@
+require './lib/main'
+
+describe Enumerable do
+  describe 'my_each' do
+    it 'should display all the numbers in an array' do
+      a = %w[sharon leo anthony]
+      expect(a.my_each { |x| x }).to eq(%w[sharon leo anthony])
+    end
+
+    it 'should return an enum if no block is given' do
+      a = %w[sharon leo anthony]
+      expect(a.my_each.class).to eq(Enumerator)
+    end
+
+    it 'should return an enum if no block is given' do
+      a = %w[sharon leo anthony]
+      expect(a.my_each.to_enum).to be_a(Enumerator)
+    end
+
+    it 'should return Class when no Enumerator and block given' do
+      a = %w[sharon leo anthony]
+      expect(a.my_each.to_enum).not_to eq(Class)
+    end
+  end
```

Comment:
Actually there is a better way to organize the test. Take a look at the following formation of code.
```ruby
describe '#my_each' do
    let(:arr) { [1, 2, 3] }
    context 'when no block is given' do
      it 'returns an enumerator' do
        expect(arr.my_each).to an_instance_of(Enumerator)
      end
      it 'returns same type value as the original each' do
        expect(arr.my_each.class).to eql(arr.each.class)
      end
    end
    context 'when a block is given' do
      it 'calls the block for each of the item in the enum and returns self' do
        expect(arr.my_each {|num| num > 5 }).to eq(arr)
      end
      it 'calls the block for each of the element in the enum as the original each' do
        each_output = 0
        arr.my_each {|num| each_output += num }
        my_each_output = each_output.dup
        each_output = 0
        arr.each {|num| each_output += num }
        expect(my_each_output).to eq(each_output)
      end
    end
end

```